scaffoldmaker/utils/tubemesh2.py

```diff
@@ -0,0 +1,356 @@
+
+def interpolatefromInnerAndOuter( xInner, xOuter, thickness, xi3, curvatureOuter, d1Outer, d2Outer, d3OuterUnit,
+    elementsCountAround, elementsCountAlong, elementsCountThroughWall):
+    """
+    Generate coordinates and derivatives at xi3 by interpolating with 
+    inner and outer coordinates and derivatives.
+    param xInner: Coordinates on inner surface
+    param xOuter: Coordinates on outer surface
+    param thickness: Thickness of wall
+    param curvatureOuter: Curvature of coordinates on outer surface
+    param d1Outer: Derivatives on outer surface around tube
+    param d2outer: Derivatives on outer surface along tube
+    param d3OuterUnit: Unit derivatives on inner surface through wall
+    param elementsCountAround: Number of elements around tube
+    param elementsCountAlong: Number of elements along tube
+    param elementsCountThroughWall: Number of elements through wall
+    return xList, dx_ds1List, dx_ds2List, dx_ds3List: Coordinates and derivatives on xi3
+    """
+    xList = []
+    dx_ds1List = []
+    dx_ds2List = []
+    dx_ds3List =[]
+
+    for n2 in range(elementsCountAlong+1):
+        for n1 in range(elementsCountAround):
+            n = n2*elementsCountAround + n1
+            norm = [-1* c for c in d3OuterUnit[n]]
+            # x
+            innerx = xInner[n]
+            outerx = xOuter[n]
+            dWall = [thickness*c for c in norm]
+            x = interpolateCubicHermite(innerx, dWall, outerx, dWall, xi3)
+            xList.append(x)
+            # dx_ds1
+            factor = 1.0 - curvatureOuter[n]*thickness*xi3
+            dx_ds1 = [ factor*c for c in d1Outer[n]]
+            dx_ds1List.append(dx_ds1)
+            # dx_ds2
+            if n2 > 0 and n2 < elementsCountAlong:
+                prevIdx = (n2-1)*elementsCountAround + n1
+                nextIdx = (n2+1)*elementsCountAround + n1
+                curvatureAround = 0.5*(
+                    getCubicHermiteCurvature(xOuter[prevIdx], d2Outer[prevIdx], xOuter[n], d2Outer[n], norm, 1.0)+
+                    getCubicHermiteCurvature(xOuter[n], d2Outer[n], xOuter[nextIdx], d2Outer[nextIdx], norm, 0.0))
+            elif n2 == 0:
+                nextIdx = (n2+1)*elementsCountAround + n1
+                curvatureAround = getCubicHermiteCurvature(xOuter[n], d2Outer[n], xOuter[nextIdx], d2Outer[nextIdx], norm, 0.0)
+            else:
+                prevIdx = (n2-1)*elementsCountAround + n1
+                curvatureAround = getCubicHermiteCurvature(xOuter[prevIdx], d2Outer[prevIdx], xOuter[n], d2Outer[n], norm, 1.0)
+
+            factor = 1.0 - curvatureAround*thickness*xi3
+            dx_ds2 = [ factor*c for c in d2Outer[n]]
+            dx_ds2List.append(dx_ds2)
+            #dx_ds3
+            dx_ds3 = [c * thickness/elementsCountThroughWall for c in norm]
+            dx_ds3List.append(dx_ds3)
+
+    return xList, dx_ds1List, dx_ds2List, dx_ds3List
+
+def rotationMatrixAboutAxis(rotAxis, theta):
```

Comment:
See earlier comment.
Whenever you find you've put the same function in 2 files you should remove it to a common file.